deeptime/markov/msm/tram/_tram.py

```diff
@@ -0,0 +1,708 @@
+            # the thermodynamic state indices.
+            return full_counts_model.submodel_largest(
+                connectivity_threshold=self.connectivity_factor,
+                directed=True)
+
+        if self.connectivity in ['post_hoc_RE', 'BAR_variance']:
+            # get state counts for each trajectory (=for each therm. state)
+            all_state_counts = np.asarray([estimator.fit_fetch(dtraj).state_histogram for dtraj in dtrajs],
+                                          dtype=object)
+            # pad with zero's so they are all the same size and easier for the cpp module to handle
+            all_state_counts = to_zero_padded_array(all_state_counts, self.n_markov_states)
+
+            # get list of all possible transitions between thermodynamic states. A transition is only possible when two
+            # thermodynamic states have an overlapping markov state. Whether the markov state overlaps depends on the
+            # sampled data and the connectivity settings and is computed in get_state_transitions:
+            if self.connectivity == 'post_hoc_RE':
+                connectivity_fn = tram.get_state_transitions_post_hoc_RE
+            else:
+                connectivity_fn = tram.get_state_transitions_BAR_variance
+
+            with CsetCallback(self.progress_bar, self.n_therm_states * self.n_markov_states) as callback:
+                (i_s, j_s) = connectivity_fn(ttrajs, dtrajs, bias_matrices, all_state_counts, self.n_therm_states,
+                                             self.n_markov_states, self.connectivity_factor, callback)
+
+            # add transitions that occurred within each thermodynamic state. These are simply the connected sets:
+            for k in range(self.n_therm_states):
+                for cset in estimator.fit_fetch(dtrajs[k]).connected_sets():
+                    i_s.extend(list(cset[0:-1] + k * self.n_markov_states))
+                    j_s.extend(list(cset[1:] + k * self.n_markov_states))
+
+            # turn the list of transitions into a boolean matrix that has a one whenever a transition has occurred
+            data = np.ones(len(i_s), dtype=int)
+            dim = self.n_therm_states * self.n_markov_states
+            sparse_transition_counts = sp.sparse.coo_matrix((data, (i_s, j_s)), shape=(dim, dim))
+
+            # Now we have all possible paths in the list of transitions. Get the connected set of that
+            overlap_counts_model = TransitionCountModel(sparse_transition_counts)
+            connected_states_ravelled = overlap_counts_model.submodel_largest(directed=False).state_symbols
+
+            # unravel the index and combine all separate csets to one cset
+            connected_states = np.unravel_index(connected_states_ravelled, (self.n_therm_states, self.n_markov_states),
+                                                order='C')
+
+            return full_counts_model.submodel(np.unique(connected_states[1]))
+
+        warnings.warn(
```

Comment:
if connectivity type (or mode) is unknown i think it is better to raise an exception initially than carry out the whole computation and warn in the end

Reply:
I would leave it as a feature. If for some reason you don't want to bother with connected sets, you can give it None as connectivity. 
I added a warning in __init__().

Reply:
In that case we should explicitly allow `None` as connectivity, document accordingly and still raise if there's a typo in there. I wouldn't want to wait for worst case hours to realize i didn't set connectivity right :-D